core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -84,16 +84,26 @@ object CityLocationTileRanker {
         tileValue += getDistanceToCityModifier(newCityTile, nearbyCities, civ)
 
         val onCoast = newCityTile.isCoastalTile()
+        val onHill = newCityTile.isHill()
+        val isNextToMountain = newCityTile.isAdjacentTo("Mountain")
         // Only count a luxary resource that we don't have yet as unique once
         val newUniqueLuxuryResources = HashSet<String>()
 
-        if (onCoast) tileValue += 8
-        if (newCityTile.isAdjacentToRiver()) tileValue += 10
+        if (onCoast) tileValue += 10
+        // Hills are free production and defence
+        if (onHill) tileValue += 7
+        // Observatories are good, but current implementation no mod-friendly
+        if (isNextToMountain) tileValue += 5
+        if (newCityTile.isAdjacentToRiver()) tileValue += 14
         if (newCityTile.terrainHasUnique(UniqueType.FreshWater)) tileValue += 5
         // We want to found the city on an oasis because it can't be improved otherwise
         if (newCityTile.terrainHasUnique(UniqueType.Unbuildable)) tileValue += 3
         // If we build the city on a resource tile, then we can't build any special improvements on it
         if (newCityTile.resource != null) tileValue -= 4
+        // We don't want to settle 1-tile islands and promontories with few good tiles
+        for (nearbyTile in newCityTile.getTilesAtDistance(1))
```

Comment:
Good idea, wrong approach. Some mods might dump way too many good resources in the water with way too high rates. Also, if we check the states of a tile, we should naturally be able to figure this out, which should also let us cover checking deserts

Reply:
> Good idea, wrong approach. Some mods might dump way too many good resources in the water with way too high rates. Also, if we check the states of a tile, we should naturally be able to figure this out, which should also let us cover checking deserts

Isn't `var rankTileValue = Automation.rankStatsValue(rankTile.stats.getTileStats(null, civ, uniqueCache), civ)` already taking care of this?

Reply:
Also, especially in the case of coast, we should look for things like the Rekmod Exploration tree and act accordingly (treasure fleets btw is meant to apply only to coast and ocean without resources).

Reply:
Well, it's not enough, as the AI *loves* settling small islands and promontories. With the change I made, it becomes somewhat less likely to do so. Alternatively, the bonus from settling on coast (which was  set to +8 by a previous dev) could be reduced.